vector-im/element-android
# Missing notifications due to stuck background workers

## Type of change


## Content

- Consumes the `lastFailureMessage` when handling the `SessionSafeCoroutineWorker.doWork()`, by not consuming, the worker is stuck in the failure state until the `WorkManager` database is cleared with a cache clean.

## Motivation and context

Fixes #6836 - implementations of `SessionSafeCoroutineWorker` suffer from uncaught exceptions causing the worker to become stuck in the failure state. 

We update the work params/input data on error but never reset the state back to the default arguments 

## Screenshots / GIFs
No UI changes

## Tests

- Add a forced uncaught crash to the `SyncWorker` 
- Install the app and attempt to receive messages whilst the app is in the background
- Remove the forced crash and attempt receive messages whilst the app is in the background
- Notice the background jobs are still failing with the original error
- `Settings -> General -> Clear cache` fixes the stuck state 

```
MatrixWorkerFactory.createWorker for org.matrix.android.sdk.internal.session.sync.job.SyncWorker
SyncWorker Params(sessionId=8742ee0c2fc165532212ce2480e7bbca, timeout=0, delay=0, periodic=false, forceImmediate=false, lastFailureMessage=java.lang.RuntimeException: manual throw)
Work cancelled due to input error from parent: java.lang.RuntimeException: manual throw
```

## Tested devices

- OS version(s): 31
